src/inspect_ai/_util/retry.py

```diff
@@ -65,3 +68,46 @@ def log_attempt(retry_state: RetryCallState) -> None:
 
 def is_httpx_connection_error(ex: BaseException) -> bool:
     return isinstance(ex, ConnectTimeout | ConnectError | ConnectionError | ReadTimeout)
+
+
+class wait_sigmoid(wait_base):
```

Comment:
If we don't have some maximum number of retries, wouldn't we run into the same indefinite hanging problem as exponential? Though I do agree this distribution seems nicer.

Reply:
Tenacity handles max_retries with a separate function (and we allow the user to set this via `--max-retries`)